translator/source_code/icone_translator.py

```diff
@@ -192,11 +322,19 @@ def parse_incident(incident):
     properties['ending_accuracy'] = "estimated"
 
     # road_name
-    properties['road_name'] = incident['location'].get('street','')
+    road_name = incident['location'].get('street', '')
+    if not road_name:
+        return None
+    properties['road_name'] = incident['location'].get('street', '')
 
     # direction
-    direction = ''
-    properties['direction'] = get_road_direction(geometry['coordinates'])
+    direction = parse_direction_from_street_name(incident['location'].get('street', ''))
+    if not direction:
+        direction = get_road_direction(geometry['coordinates'])
+        if not direction:
+            return None
```

Comment:
Is this filtering out all single point messages? IE if an iCone message has two points and they are the same location, would that get filtered out here with this `return None`? I think those messages would still be valid.

Reply:
Yes, we are ignoring incidents with no direction or road name information.Both of the fields are required in the WZDx message and if we can not determine the road name or the road direction (N,E,S,W), then a WZDx message cannot be created for that incident.

Reply:
Closing for now, but I think we'll want to look at this logic a bit closer in the next sprint. We could easily have a valid sign on the side of the road that would have the same start/end points since its just a single location. I'm not sure this would have a direction or not, but if not we'll have to determine how to best handle it.